crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4158,6 +4159,40 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn resolve_unbound_public() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            "
+            if flag:
+                x: int = 1
+            ",
+        )?;
+
+        assert_public_ty(&db, "src/a.py", "x", "int");
+
+        Ok(())
+    }
+
+    #[test]
+    fn resolve_unannotated_unbound_public() -> anyhow::Result<()> {
```

Comment:
nit: it's only maybe unbound
```suggestion
    fn resolve_unannotated_maybe_unbound_public() -> anyhow::Result<()> {
```